scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -0,0 +1,544 @@
+Then we can subset the pseudo-bulked object to just contain our cell type of interest before proceeding. 
+
+Before we apply pseudo-bulking to our dataset, lets look at a simple example of how pseudo-bulking works. 
+
+```{r}
+# create an example counts matrix
+counts_mtx <- matrix(1:12, 
+                     ncol = 4,
+                     dimnames = list(c("geneA", "geneB", "geneC"),
+                                     c("A-cell1", "A-cell2", "B-cell1", "B-cell2")))
+counts_mtx
+```
+
+
+```{r}
+# create new matrix, summing the counts for each gene across all cells in the specified group
+pseudobulked_counts <- cbind(
+  rowSums(counts_mtx[,1:2]), # sum all counts for cells labeled with group A
+  rowSums(counts_mtx[,3:4]) # sum all counts for cells labeled with group B
+)
+pseudobulked_counts 
+```
+Now the actual pseudo-bulking for our dataset! 
+We can pseudo-bulk using any groups that we are interested in.
+For right now, we would are interested in looking at gene expression across cell types, so we want to group the counts matrix by both cell type and sample. 
+
+```{r}
+# first let's subset our coldata to only have the columns we care about in pseudobulking 
+pseudobulked_coldata <- colData(rms_sce)[, c("celltype_broad", "sample")]
+
+# this creates a new SCE object that contains the pseudobulked counts across the provided groups 
+pseudobulked_sce <- scater::aggregateAcrossCells(rms_sce, 
+                                                  id = pseudobulked_coldata)
+
+# column names aren't automatically added, so let's add them in 
+colnames(pseudobulked_sce) <- paste(pseudobulked_sce$celltype_broad, 
+                                    pseudobulked_sce$sample, sep = "-")
+
+pseudobulked_sce
+```
+
+How does the new pseudo-bulked `SingleCellExperiment` look different? 
+
+```{r}
+# only 37 columns in the counts assay 
+counts(pseudobulked_sce)[1:10, 1:10]
+```
+
+Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+
+```{r}
+# note the new addition of the column with number of cells per group 
+colData(pseudobulked_sce)
+```
+
+### Filtering the pseudo-bulked `SingleCellExperiment` object
+
+*Add a bit more justification on filtering and why we need it*
+
+```{r}
+# filter to only the HVG list
+hv_genes <- metadata(integrated_sce)$combined_hvg
+filtered_pseudobulked_sce <- pseudobulked_sce[hv_genes, ]
+```
+
+We want to remove any groups in the pseudo-bulked dataset that may have a low number of cells. 
+
+```{r}
+# filter by number of cells 
+filtered_pseudobulked_sce <- filtered_pseudobulked_sce[, pseudobulked_sce$ncells >= 10]
+```
+
+We also want to filter out any genes that may be lowly expressed in our pseudo-bulked dataset.
+
+*Talk about how this specific function uses the sample groups provided to determine the minimum sample size that genes are expressed in.*
+
+```{r}
+# filter out genes that might be lowly expressed using filterbyExpr in `edgeR`
```

Comment:
might want to note here that we're not actually using edgeR for the differential expression but it has useful utility functions